GP11 : DSA-SHA224 DSA-SHA256 algorithms

cedric-chaumont-st-dev opened the pull request.

jenswi-linaro commented on host/xtest/Makefile:
```diff
@@ -10,6 +10,10 @@ endif
 
+# If CFG_XTEST_LITE=n : compile full xtest suite (default)
+# If CFG_XTEST_LITE=y : compile sanity xtest suite
+CFG_XTEST_LITE ?= n
```
Runtime configuration would be more useful.

cedric-chaumont-st-dev replied:
Do you mean something like CFG_RUNTIME_LITE ?

cedric-chaumont-st-dev replied:
Do you mean something like "xtest lite" ?  Could it be done in another PR ?

jforissier replied:
+1 with @jenswi-linaro. IMO compile-time options are inconvenient, so they should not be used unless there is a reason (like reducing file size, or incompatible options...).
It is just a matter of replacing `#ifndef ... #endif` with `if (...) { ... }`, no?

pascal-brand-st-dev replied:
+1 with @jenswi-linaro 
But this is not just a `if...` as the current `ifndef` is in the definition of the testcases to take into account. One solution is to add a boolen in `XTEST_AC_DSA_CASE` to indicate if the test case should be run in the light suite, and then add the `if` (same for #24)

So I would vote to have it in another PR (soon hopefully?) and to remove `CFG_XTEST_LITE` from the current PR

jforissier replied:
It can be a simple `if` I think: what about adding  global variable (`option_light` for instance) that would be set when the command line is parsed, and could be tested from anywhere? Personally I don't like the idea of adding things when we know we want to remove them ASAP.

jenswi-linaro replied:
I agree with Pascal, let's do the "lite" stuff in a separate PR.

cedric-chaumont-st-dev commented:
Updated. Squashed. Verified.
+-----------------------------------------------------
37980 subtests of which 1 failed
687 test cases of which 1 failed
0 test case was skipped
TEE test application done!

NOTE: GP test 7671 failing (patch on going)

pascal-brand-st-dev commented:
Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

cedric-chaumont-st-dev commented:
Update git comment.

pascal-brand-st commented:
Merged manually
